src/lib/operators/join_hash/join_hash_steps.hpp

```diff
@@ -557,29 +582,38 @@ void probe_semi_anti(const RadixContainer<RightType>& radix_container,
 using PosLists = std::vector<std::shared_ptr<const PosList>>;
 using PosListsBySegment = std::vector<std::shared_ptr<PosLists>>;
 
+/**
+ * Returns a vector where each entry with index i references a PosLists object. The PosLists object
+ * contains the position list of every segment/chunk in column i.
+ * @param input_table
+ */
 // See usage in _on_execute() for doc.
 inline PosListsBySegment setup_pos_lists_by_segment(const std::shared_ptr<const Table>& input_table) {
   DebugAssert(input_table->type() == TableType::References, "Function only works for reference tables");
 
+  // Why do we need this map?
```

Comment:
Please make another pass over this PR again yourself. There are several comments that suggest that you are not done with it.